src/lib.rs

```diff
@@ -966,14 +965,7 @@ impl<'a, T> Send<'a, T> {
             let msg = self.msg.take().unwrap();
             // Attempt to send a message.
             match self.sender.try_send(msg) {
-                Ok(()) => {
```

Comment:
Why did you get rid of this one?

Reply:
Hmm, not totally sure what your question is about.
The same thing is done for the `Recv` future, and this is because the original mechanism was meant to implement a "slow" notification chain to limit the number of simultaneous in-flight notifications. So only one notification was sent at a time, but this made it necessary to unconditionally send another notification, whether this notification was needed or not.
This is no longer necessary since now all notifications are sent, whether or not one is already in-flight.
Note that PR #16 did the exact same thing and the original author seemed to agree it was correct. And all tests pass, which is another indicator that it should be OK :-)